doc/swss-schema.md

```diff
@@ -620,7 +622,42 @@ Equivalent RedisDB entry:
     12) "0"
     127.0.0.1:6379>
 
-### Configuration files
+
+## Configuration DB schema
+
+### WARM\_RESTART
+    ;Stores system warm start configuration
+    ;Status: work in progress
+
+    key             = WARM_RESTART:name         ; name is the name of SONiC docker or "system" for global configuration.
+
+    enable          = "true" / "false"          ; Default value as false.
+                                                ; If "system" warm start knob is true, docker level knob will be ignored.
+                                                ; If "system" warm start knob is false, docker level knob takes effect.
```

Comment:
docker knob always override system level knob, that was the use cases we dicussed.

Reply:
The tricky point is the default value is false.   If docker knob always override system level knob,  when warm restart is enabled at system level, each docker has to be enabled too. 

What is the use case of having warm restart enabled at system level, while disabled for certain docker?

Reply:
if docker knob does not have this attribute, then use system level.

the use case was rodney's, snmp docker snmp warm reboot is available but experimental, so in production he wants to explicitly disable it although system level is enabled.

Reply:
@rodnymolina  for snmp case, probably it should alway do cold boot  as other dockers like lldp, dhcp_relay until its warm reboot support is stable enough.  

If the existence of certain attribute is treated as one condition, then it becomes tri-state variable:  true/false/not-exist,  to me, it seems error prone. We don't have such case in sonic yet(?).

Reply:
In general, more specific knobs take precedence over global configuration. I think we can use this principle here as well.
If we don't like the not-exist case, we need this knob for every docker. For the time-being, we can define the knobs for dockers which have code to support warm reboot, for the ones do not support warm reboot, they do not look at the knobs anyways.

Reply:
If docker knob always takes precedence over system knob,  the not-exist case is asked implicitly, otherwise why system knob is needed if it gets preempted all the time. That means the default value for enable is not-exist.

The knobs are also used to control whether to flush DB.  
Ex. based on the current schema,  if system knob or one docker knob enabled, we keep the appDB and some other DBs data. 

But if docker knob always takes precedence,  and it is disabled,  should the DB be flushed or not.   If DB is not flushed, the specific docker will have to handle the scenarios of DB with existing data though it is supposed to do a cold start.   If DB is flushed, how about other dockers?

Better not to support the case of system knob enabled, while docker knob disabled and takes effect.  That is one of the reasons for  `If "system" warm start knob is true, docker level knob will be ignored.`

Reply:
@jipanyang  As talked, this "system" knob was meant for whole system warm reboot, not a global setting of docker restart, that would lead to a different discussion.

Could you please help to provide (the current) work flow of using these knobs during the shutdown and bringing up of the dockers/system? We can then think whether the whole work flow make sense and keeps everybody on the same page.